GiniVision/Classes/ReviewContainerViewController.swift

```diff
@@ -78,10 +77,54 @@ internal class ReviewContainerViewController: UIViewController, ContainerViewCon
         displayContent(contentController)
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        setupLeftBarButton()
+    }
+    
+    fileprivate func setupLeftBarButton() {
+        guard let navController = navigationController else {
+            return
+        }
+        
+        // Configure back button. Needs to be done here because otherwise the navigation controller would be nil
+        if navigationItem.leftBarButtonItem == nil {
+            setupLeftNavigationItem(usingResources: backButtonPreferredResource(forNavController: navController), selector:#selector(back))
+        }
+    }
+    
     @IBAction func back() {
         let delegate = (navigationController as? GiniNavigationViewController)?.giniDelegate
         delegate?.didCancelReview?()
-        _ = navigationController?.popViewController(animated: true)
+        
+        handleBackPressed()
+    }
+    
+    fileprivate func handleBackPressed() {
+        guard let navController = navigationController else {
+            return
+        }
+        
+        if self.firstViewController(inNavController: navController) {
+            navController.dismiss(animated: true, completion: nil)
+        } else {
+            _ = navController.popViewController(animated: true)
+        }
+    }
+    
+    fileprivate func backButtonPreferredResource(forNavController navController:UINavigationController) -> PreferredButtonResource {
+        if self.firstViewController(inNavController: navController) {
+            return closeButtonResources
+        }
+        return backButtonResources
+    }
+    
+    fileprivate func firstViewController(inNavController navController:UINavigationController ) -> Bool {
```

Comment:
A name like `firstViewController ` makes me think it'd return a view controller. Isn't `isFirstViewController` better and more swifty?